UPGRADE.md

```diff
@@ -1,5 +1,10 @@
 # Upgrade to 2.11
 
+## Deprecated `MasterSlaveConnection`
+
+The naming is offensive, the implementation very questionable. Use regular
+connections and switch between them explicitely.
```

Comment:
Should it be “explicitly”? I'd also avoid “switch”. Let's use “Use each endpoint connection explicitly”.